lib/deploy/lifecycleHooks.js

```diff
@@ -1,43 +1,78 @@
 'use strict';
 
+var _ = require('lodash');
 var RSVP = require('rsvp');
 var utils = require('../utils');
 var chalk = require('chalk');
-var process = require('child_process');
+var cprocess = require('child_process');
 var FirebaseError = require('../error');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+
+
+function runCommand(command, options) {
+  return new Promise(function(resolve, reject) {
+    var child = cprocess.spawn(command, _.assign({}, options, {
+      shell: true,
+      stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
+    })
+    );
+    logger.info('Running command: ' + command);
+    console.log(process.env);
+    child.on('error', function(err) {
+      reject(new Error(err));
+    });
+    child.on('exit', function(code, signal) {
+      if (signal !== null) {
+        reject(new Error('Command terminated with signal ' + signal));
+      } else if (code) {
+        reject(new Error('Command terminated with non-zero exit code' + code));
+      } else {
+        resolve();
+      }
+    });
+  });
+}
 
 module.exports = function(target, hook) {
   // Errors in postdeploy script will not exit the process since it's too late to stop the deploy.
   var exit = hook !== 'postdeploy' ? undefined : {exit: 2};
 
   return function(context, options) {
-    var command  = options.config.get(target + '.' + hook);
-    if (!command) {
+    var commands  = options.config.get(target + '.' + hook);
+
+    if (!commands) {
       return RSVP.resolve();
     }
+    if (typeof commands === 'string') {
+      commands = [commands];
+    }
+    // active project ID
+    process.env.GCLOUD_PROJECT = getProjectId(options);
```

Comment:
Please don't change the current process' environment variables; this might create odd bugs. Instead, you can use the [`env` option in spawn](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options).

Reply:
Also, I'm on the fence whether the env parameter should be `_.assign({}, process.env, <custom>)` or whether we should avoid leaking unknown env variables that won't be there in production. WDYT @laurenzlong ?